server.go

```diff
@@ -313,6 +313,11 @@ type Server struct {
 	// Shutdown handling
 	lock    sync.RWMutex
 	started bool
+	// track packet processing and allow to wait all are processed before closing
+	dnsWg       sync.WaitGroup
+	// Shutdown timeout. If not indicated, the timeout will be max(readTimeout, idleTimeout) + 1 sec
+	ShutdownTimeout time.Duration
```

Comment:
I don't think we need or want *another* timer in this struct, just us the readTimeout, idleTimeout .

Reply:
ok.